update_translations.py

```diff
@@ -0,0 +1,116 @@
+
+# name of the master hq translations file to be updated
+hq_translations_filename = 'messages_en-2.txt'
+
+# relative path to the subfolder within the javarosa repo containing the messages_default.txt file
+javarosa_subfolder = './j2me/shared-resources/resources'
+
+# relative path to the subfolder within the commcare repo containing the messages_cc_default.txt file
+commcare_subfolder = './application/resources'
+
+# relative path to the subfolder within the commcare-odk repo containing the messages_ccodk_default.txt file
+ccodk_messages_subfolder = './app/assets/locales'
+
+# relative path to the subfolder within the commcare-odk repo containing the strings.xml file
+ccodk_strings_subfolder = './app/res/values'
+
+javarosa_repo = 'javarosa'
+commcare_repo = 'commcare'
+commcare_odk_repo = 'commcare-odk'
+translations_repo = 'commcare-translations'
+
+javarosa_filename = 'messages_default.txt'
+commcare_filename = 'messages_cc_default.txt'
+ccodk_messages_filename = 'messages_ccodk_default.txt'
+ccodk_strings_filename = 'strings.xml'
+
+javarosa_header_string = '# *** messages_default.txt ***'
+commcare_header_string = '# *** messages_cc_default.txt ***'
+ccodk_messages_header_string = '# *** messages_ccodk_default.txt ***'
+ccodk_strings_header_string = '# *** strings.xml ***'
+
+header_strings = [javarosa_header_string, commcare_header_string, ccodk_messages_header_string, ccodk_strings_header_string]
+
+strings_namespace = '{http://strings_namespace}'
+
+
+def update_translations(new_version_number):
+    new_javarosa_text = get_updated_translations(javarosa_repo, javarosa_subfolder, javarosa_filename)
+    new_commcare_text = get_updated_translations(commcare_repo, commcare_subfolder, commcare_filename)
+    new_ccodk_text = get_updated_translations(commcare_odk_repo, ccodk_messages_subfolder, ccodk_messages_filename)
+    new_strings_text = get_updated_strings_block()
+    new_text_blocks = [new_javarosa_text, new_commcare_text, new_ccodk_text, new_strings_text]
+
+    new_branch_name = checkout_new_translations_branch(new_version_number)
+    backup_old_translations_file()
+    create_updated_translations_file(new_text_blocks)
+    commit_and_push_new_branch(new_version_number, new_branch_name)
+
+
+def checkout_new_translations_branch(new_version_number):
+    chdir_repo(translations_repo)
+    subprocess.call('git checkout master', shell=True)
+    subprocess.call('git pull origin master', shell=True)
+    new_branch_name = '{}_release_additions'.format(new_version_number)
+    subprocess.call('git checkout -b {}'.format(new_branch_name), shell=True)
+    return new_branch_name
+
+
+def backup_old_translations_file():
+    subprocess.call('mv {0} {1}'.format(hq_translations_filename, hq_translations_filename + '.bak'), shell=True)
+
+
+def create_updated_translations_file(new_text_blocks):
+    subprocess.call('touch {}'.format(hq_translations_filename), shell=True)
+    f = open(hq_translations_filename, 'w')
+    for i in range(4):
```

Comment:
Picky, but this is a bit of a magic number, I'd replace with `len(header_strings)` or replace the loop with `for i, header in enumerate(header_strings):` (and would change the `< 3` below for the same reason).